src/WebJobs.Extensions/Extensions/Timers/Scheduling/CronSchedule.cs

```diff
@@ -116,38 +164,40 @@ private DateTimeOffset GetNextOccurrence(DateTimeOffset now)
+                // "Fall Back" Scenario 1 -- Point-in-time schedule where the next time is ambiguous and the offsets have changed.
                 //   This means that "now" is in DST and "next" is in Standard Time. Since we never want to run an ambiguous
                 //   point-in-time schedule on Standard time, move the offset back to the Daylight offset.
-                var offsetDiff = nextOffset - now.Offset;
-                next = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), timeZone);
+                var offsetDiff = next.Offset - now.Offset;
+                adjusted = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), timeZone);
             }
-            else if (!IsInterval && isNextAmbiguous && nextOffset == now.Offset)
+            else if (!IsInterval && isNextAmbiguous && next.Offset == now.Offset)
             {
-                // Scenario 3 -- Point-in-time where "next" is ambiguous and "now" and "next" are in Standard time. This can happen
+                // "Fall Back" Scenario 2 -- Point-in-time where "next" is ambiguous and "now" and "next" are in Standard time. This can happen
                 //   when the timer starts and we are already past the "fall back" point. For example, if the trigger starts up at
                 //   01:29-8 and we have a "every day at 01:30" schedule, we have to assume that we've already run it at 01:30-7 and
                 //   therefore calculate the time after this.
+                var nextDateTime = next.LocalDateTime;
+
                 while (timeZone.IsAmbiguousTime(nextDateTime))
                 {
                     nextDateTime = _cronSchedule.GetNextOccurrence(nextDateTime);
                 }
 
-                next = new DateTimeOffset(nextDateTime, timeZone.GetUtcOffset(nextDateTime));
+                adjusted = new DateTimeOffset(nextDateTime);
             }
-            else if (IsInterval && (isNowAmbiguous || isNextAmbiguous) && nextOffset != now.Offset)
+            else if (IsInterval && (isNowAmbiguous || isNextAmbiguous) && next.Offset != now.Offset)
             {
-                // Scenario 4 -- Interval schedule where either point is ambiguous and we're crossing the offset boundary. For example, 
+                // "Fall Back" Scenario 3 -- Interval schedule where either point is ambiguous and we're crossing the offset boundary. For example, 
                 //   an "every 30 minute" interval in Pacific time:
                 //   - 00:30-7 -> 01:00-8 (because all ambiguous "next" use Standard time) -> adjust back 01:00-7
                 //   - 01:30-7 -> 02:00-8 -> adjust back to 01:00-8
-                var offsetDiff = nextOffset - now.Offset;
-                next = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), timeZone);
+                var offsetDiff = next.Offset - now.Offset;
+                adjusted = TimeZoneInfo.ConvertTime(next.Add(offsetDiff), timeZone);
             }
 
-            return next;
+            return adjusted != null;
```

Comment:
nit: I'd like to see us using the `is` operator more throughout our repos for improved code readability and consistency

```suggestion
            return adjusted is not null;
```